### PR TITLE
submit js3 week1 solutions- Abdullah

### PR DESCRIPTION
**Pull request by abdullahkanatsiz7**
my js3 week1 homework

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -28,27 +28,74 @@
     });
     return elem;
   }
+  function NewDate(dateString) {
```
Comment (remarcmij) [inline homework/index.js]: Function, variable and parameter names should be in `camelCase`: `newDate`. Moreover, `newDate` is not the best possible name. You are not creating a new date, you are just formatting an existing date for presentation purposes. Therefore, the name `formatDate` would seem a better fit.

File: homework/index.js [hunk 2]
```diff
@@ -28,27 +28,74 @@
...
+    const dateTime = new Date(dateString);
+    return dateTime.toLocaleString();
+  }
+
+  function renderRepoDetails(repo, section) {
+    const Mix = createAndAppend('ul', section);
```
Comment (remarcmij) [inline homework/index.js]: This variable name should start with a lowercase letter. What does `mix` mean in this context?

File: homework/index.js [hunk 3]
```diff
@@ -28,27 +28,74 @@
...
+    createAndAppend('p', description, {
+      text: repo.description,
+      class: 'right-side-2',
+    });
+    createAndAppend('p', fork, { text: repo.forks, class: 'right-side-3' });
+    createAndAppend('p', time, {
+      text: NewDate(repo.updated_at),
+      class: 'right-side-4',
+    });
+  }
```
Comment (remarcmij) [inline homework/index.js]: The repository details that you render look like a table with two columns and four rows. Therefor an HTML table would be a natural choice. I noticed that you have manually fine-tuned the left padding in the classes `right-side-n` to vertically line up the values in the second column. However, this is not the recommended way as it is depended on the chosen font etc., which can lead to maintenance difficulties in the future when somebody decides the change the chosen font.

If you look closely on the UI as displayed on a mobile phone you can already see that the second column is not accurately aligned vertically:

image

Using an HTML table will solve this:

image

If this was an app on the App Store, which one would you install?

File: homework/index.js [hunk 4]
```diff
@@ -28,27 +28,74 @@
...
           text: err.message,
           class: 'alert-error',
         });
         return;
       }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+      const section = createAndAppend('section', root);
+
+      repos.sort(sortName).forEach(repo => renderRepoDetails(repo, section));
```
Comment (remarcmij) [inline homework/index.js]: Since the function body of `sortName` is only a single line, I would prefer to code this function in-line as an arrow function:

```js
repos
  .sort((a, b) => a.name.localeCompare(b.name))
  .forEach(repo => renderRepoDetails(repo, repoContainer));
```

File: homework/index.js [hunk 5]
```diff
@@ -28,27 +28,74 @@
...
     });
   }
-
-  const HYF_REPOS_URL =
+  const HYF_REPO_URL =
```
Comment (remarcmij) [inline homework/index.js]: Was this change really necessary?

File: homework/index.js (hunk 1 above)
Comment (abdullahkanatsiz7) [reply homework/index.js]: Firstly Thank you for this feedback. Yes, you right about that. I'm going to change its name

File: homework/index.js (hunk 2 above)
Comment (abdullahkanatsiz7) [reply homework/index.js]: I think list name will be better than mix

File: homework/index.js (hunk 4 above)
Comment (abdullahkanatsiz7) [reply homework/index.js]: thank you for this information because this is more short way

File: homework/index.js (hunk 5 above)
Comment (abdullahkanatsiz7) [reply homework/index.js]: of course not, I only forget that make change

File: homework/index.js (hunk 3 above)
Comment (abdullahkanatsiz7) [reply homework/index.js]: now I changed my function

File: homework/index.js [hunk 6]
```diff
@@ -44,10 +70,14 @@
         return;
       }
       const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+      const li = createAndAppend('li', ul);
+
+      repos
+  .sort((a, b) => a.name.localeCompare(b.name))
+  .forEach(repo => renderRepoDetails(repo, li));
```
Comment (remarcmij) [inline homework/index.js]: You are using a single list item that contains all the repositories. I would expect one list item per repository. So each `table` element should be wrapped in an `li`.

Comment (abdullahkanatsiz7) [reply homework/index.js]: Yes I reviewed another project for find table solution. I know this is not good for me. If you want  I  work to find another solution

Comment (remarcmij) [reply homework/index.js]: No, you don't need to find another solution as far as I'm concerned. But for a next time, please try for yourself first and ask help if you get stuck.

Comment (abdullahkanatsiz7) [reply homework/index.js]: yes you right I need to do like your said. Next time I will use only my solution. Thank you for this good feedback
